File: 99_afleveringer/01_file_formats_bonanza/python/main.py
from parser_classes import BookParser
from data_classes import BookData

if __name__ == '__main__':
    print('-----TXT-----')
    print(BookParser.import_book_from_txt('99_afleveringer/01_file_formats_bonanza/files/book.txt').to_string())

    print('-----CSV-----')
    print(BookParser.import_book_from_csv('99_afleveringer/01_file_formats_bonanza/files/book.csv').to_string())

    print('-----JSON-----')
    print(BookParser.import_book_from_json('99_afleveringer/01_file_formats_bonanza/files/book.json').to_string())

    print('-----XML-----')
    print(BookParser.import_book_from_xml('99_afleveringer/01_file_formats_bonanza/files/book.xml').to_string())

    print('-----YAML-----')
    print(BookParser.import_book_from_yaml('99_afleveringer/01_file_formats_bonanza/files/book.yaml').to_string())

File: 99_afleveringer/01_file_formats_bonanza/java/src/main/java/Main.java
// Main class for testing purposes
public class Main {
    public static void main(String[] args) {
        // Test CSV import
        System.out.println("-----CSV-----");
        System.out.println("" + Parser.importBooksFromCSV("99_afleveringer/01_file_formats_bonanza/files/book.csv"));

        // Test JSON import
        System.out.println("-----JSON----");
        System.out.println("" + Parser.importBookFromJSON("99_afleveringer/01_file_formats_bonanza/files/book.json"));

        // Test YAML import
        System.out.println("-----YAML----");
        System.out.println("" + Parser.importBookFromYAML("99_afleveringer/01_file_formats_bonanza/files/book.yaml"));

        // Test XML import
        System.out.println("-----XML-----");
        System.out.println("" + Parser.importBookFromXML("99_afleveringer/01_file_formats_bonanza/files/book.xml"));

        // Test TXT import
        System.out.println("-----TXT-----");
        System.out.println("" + Parser.importBookFromTXT("99_afleveringer/01_file_formats_bonanza/files/book.txt"));
    }
}